src/bmi/bmi_sac.f90

```diff
@@ -1153,16 +1191,32 @@ function sac_set_int(this, name, src) result (bmi_status)
     character (len=*), intent(in) :: name
     integer, intent(in) :: src(:)
     integer :: bmi_status
+    integer(kind=int64) :: exec_status
 
     !==================== UPDATE IMPLEMENTATION IF NECESSARY FOR INTEGER VARS =================
 
     select case(name)
 !     case("model__identification_number")
 !        this%model%id = src(1)
 !        bmi_status = BMI_SUCCESS
-    case default
-       bmi_status = BMI_FAILURE
-       call write_log(" Failed to set integer value for  " // name // "", LOG_LEVEL_WARNING)
+      case("serialization_create")
+         call new_serialization_request(this%model, exec_status)
+         if (exec_status == 0) then
+            bmi_status = BMI_SUCCESS
+         else
+            bmi_status = BMI_FAILURE 
+         end if
+      case("serialization_state")
+         call deserialize_mp_buffer(this%model)
```

Comment:
There's nothing in the deserialization path that actually sets the contents of `this%model%serialization_buffer` from which messagepack is going to try to unpack it.